### PR TITLE
Add PodSecurityPolicyTemplate 

### LINKED ISSUE #244: Add PSP Templates
WIP https://github.com/axeal/terraform-provider-rancher2/tree/add-psp

### PR DESCRIPTION
**Pull request by zparnold**
This adds:

- PodSecurityPolicyTemplate resource
- PodSecurityPolicyTemplate datasource
- PodSecurityPolicyTemplate schema
- PodSecurityPolicyTemplate structure
- PodSecurityPolicyTemplate docs

Acceptance tests are working

If possible could a release be cut as soon as this is merged, our company would like to use this ASAP

Fixes: #244

### CODE REVIEW

**Comment by zparnold** [comment]
@rawmind0 @axeal Not sure if you guys are the right people to ping on this, but would love feedback.

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ bin/
 build/
 dist/
 */*.swp
+vendor/
```
Comment (rawmind0) [inline .gitignore]: Why this change??  vendor folder is updated and maintained in the git repo.

File: rancher2/structure_pod_security_policy_run_as_user_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+			testPodSecurityPolicyRunAsUserInterface,
+		},
+		{
+			testNilPodSecurityPolicyRunAsUserConf,
+			testEmptyPodSecurityPolicyRunAsUserInterface,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Logf("Expected: %#v\nGiven:    %#v", tc.Input, tc.ExpectedOutput)
```
Comment (rawmind0) [inline rancher2/structure_pod_security_policy_run_as_user_test.go]: Could you please remove this log entry??

File: go.mod [hunk 3]
```diff
@@ -3,7 +3,6 @@ module github.com/terraform-providers/terraform-provider-rancher2
 go 1.12
 
 require (
-	github.com/google/btree v1.0.0 // indirect
```
Comment (rawmind0) [inline go.mod]: Why this change??

**Review by rawmind0 — CHANGES_REQUESTED** [review]
@zparnold , thanks for the cool PR.

Could you please take a look to comments??

File: go.mod (hunk 3 above)
Comment (zparnold) [reply go.mod]: My IDE removed it I can add it back

File: .gitignore (hunk 1 above)
Comment (zparnold) [reply .gitignore]: Whoops didn't see that 😂

File: rancher2/structure_pod_security_policy_run_as_user_test.go (hunk 2 above)
Comment (zparnold) [reply rancher2/structure_pod_security_policy_run_as_user_test.go]: Yes

**Comment by zparnold** [comment]
@rawmind0 Addressed PR comments. And warmly await your reply 😄

**Review by rawmind0 — APPROVED** [review]

**Review by rawmind0 — APPROVED** [review]
Thanks @zparnold 

LGTM

**Comment by rawmind0** [comment]
@zparnold , we have another PR's in flight. We expect to cut a new release next week.

**Comment by zparnold** [comment]
Thanks! Have a great weekend! :tada:
